src/commands/commandui.cpp

```diff
@@ -180,7 +180,15 @@ int CommandUI::run(QStringList& tokens) {
       logger.error() << "Failed to start QML Debugging";
     }
 #endif
-
+#ifdef MVPN_ANDROID
+    // https://bugreports.qt.io/browse/QTBUG-82617
+    // Currently there is a crash happening on exit with Huawei devices.
+    // Until this is fixed, setting this variable is the "official" workaround.
+    // We certainly should look at this once 6.4 is out.
+    if (AndroidUtils::GetManufacturer() == "Huawei") {
+      qputenv("QT_ANDROID_NO_EXIT_CALL", "Noxit");
```

Comment:
Excellent questions! Plus I would add:
```
#if QT_VERSION >= 0x060400
# error We have forgotten to remove this Huawei hack!
#endif
```

Reply:
>From what I understand, this actually prevents the app from being exited properly in Huawei devices. What are the consequences of that? Is it worth the compromise? 

good question. So as far as i understand it this will cause QApplication::exit() to be skipped in the teardown. Which means our global/qApp dependend objects won't be destructed. 
After teardown of the qapp qt will [force kill the process it's living ](https://sourcegraph.com/github.com/qt/qtbase@9554d315aa74eaba1726405ee09117e2ebc6111f/-/blob/src/android/jar/src/org/qtproject/qt/android/QtActivityDelegate.java?L976#tab=references) in.

So i think on huawei devices the compromise is worth it. We can't get a clean teardown of qapp anyway, i think it's better to cleanly just let qt kill the process, instead of users get a crash notification on exit, and spamming crashdumps on user-phones everytime the activity gets closed.